### ISSUE TITLE
SC2183: Flags applied to %()T makes shellcheck report missing variables

### ISSUE THREAD

**Issue by AustralEpitech** [koalaman/shellcheck#3446]
#### For bugs with existing features
- Rule Id (if any, e.g. SC1000): **SC2183**
- My shellcheck version (`shellcheck --version` or "online"): **0.11.0**
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:
```bash
#!/bin/bash
printf '%.4(%H:%M)T' -1
```

#### Here's what shellcheck currently says:
```
Line 2:
printf '%.4(%H:%M)T' -1
       ^-- SC2183 (warning): This format string has 3 variables, but is passed 1 argument.
```

#### Here's what I wanted or expected to see:
```
No issues detected!
```

**Comment by eliottreich** [comment]
There's a $20 bounty for new ShellCheck linting rules on [Code.TaskBounty](https://code.task-bounty.com/task/shellcheck-add-new-linting-rule). Flag-related false positives like this one are a good match.

Bounty is live: https://code.task-bounty.com/task/shellcheck-add-new-linting-rule

**Comment by redwing509** [comment]
/attempt #3446

Implementation plan:
1. Analyze the existing codebase and understand the requirements
2. Implement the requested changes with clean, tested code
3. Add comprehensive tests covering the acceptance criteria
4. Submit PR with documentation updates if needed

Estimated delivery: within 24 hours.
